### ISSUE TITLE
Version 3.4.0 seems to break iOS local build

### ISSUE THREAD

**Issue by Eadinator** [expo/eas-cli#1648]
### Summary

Hi

We have a project that we are only building for iOS currently, on our own hardware.

The command we use is `EXPO_TOKEN="<redacted>" npx eas build --platform ios --non interactive --local`

The builds completed successfully on eas-cli version 3.3.2. The same commits that built on 3.3.2 do not build on version 3.4.0 however.

The final few build log lines in 3.4.0 are as follows.

`21-Jan-2023 21:31:55 	Build successful
21-Jan-2023 21:31:55 	You can find the build artifacts in <redacted>.ipa
21-Jan-2023 21:32:00 	
21-Jan-2023 21:32:00 	- Compressing project files
21-Jan-2023 21:32:00 	
21-Jan-2023 21:32:00 	- Compressing project files
21-Jan-2023 21:32:00 	- Waiting for builds to complete. You can press Ctrl+C to exit.
21-Jan-2023 21:32:00 	- Waiting for builds to complete. You can press Ctrl+C to exit.
21-Jan-2023 21:32:00 	    Error: Build for platform ANDROID must be defined in this context`

I'm happy to provide more details as required.

Thanks!

### Managed or bare?

Managed

### Environment

N/a

### Reproducible demo or steps to reproduce from a blank project

N/a

**Comment by hoangdaicntt** [comment]
I tried to install 3.3.2 but the error is still there

**Comment by hoangdaicntt** [comment]
@Eadinator Can you revert to the previous version?

**Comment by Eadinator** [comment]
> @Eadinator Can you revert to the previous version?

Yes, reverting to 3.3.2 resolves the issue but it is not clear why 3.4.0 displays this error. There are no breaking changes in the 3.4.0 changelog.

**Comment by mlsm-trl** [comment]
I'm also facing this error, but when trying to build for Android instead, and I can confirm rollback to 3.3.2 works

![image](https://user-images.githubusercontent.com/11164660/213935929-1840322c-8c9c-46fe-8a91-2f157437f43f.png)

**Comment by dsokal** [comment]
This is fixed in https://github.com/expo/eas-cli/releases/tag/v3.4.1

**Comment by Eadinator** [comment]
Great, thanks
